internal/hook/hook_tracker.go

```diff
@@ -48,37 +48,42 @@ type hookTrackerKey struct {
 	container string
 }
 
-// hookTrackerVal records the execution status of a specific hook.
-// hookTrackerVal is extensible to accommodate additional fields as needs develop.
-type hookTrackerVal struct {
+// hookStatus records the execution status of a specific hook.
+// hookStatus is extensible to accommodate additional fields as needs develop.
+type hookStatus struct {
 	// HookFailed indicates if hook failed to execute.
 	hookFailed bool
 	// hookExecuted indicates if hook already execute.
 	hookExecuted bool
+	// hookErr records error if hook execution fails
+	hookErr error
 }
 
 // HookTracker tracks all hooks' execution status
+// HookTracker is a map that uses the backup/restore name as the key
+// and stores another map representing a group of hooks associated with a specific backup/restore.
 type HookTracker struct {
 	lock    *sync.RWMutex
-	tracker map[hookTrackerKey]hookTrackerVal
+	tracker map[string]map[hookKey]hookStatus
 }
 
 // NewHookTracker creates a hookTracker.
 func NewHookTracker() *HookTracker {
 	return &HookTracker{
 		lock:    &sync.RWMutex{},
-		tracker: make(map[hookTrackerKey]hookTrackerVal),
+		tracker: make(map[string]map[hookKey]hookStatus),
 	}
 }
 
 // Add adds a hook to the tracker
 // Add must precede the Record for each individual hook.
 // In other words, a hook must be added to the tracker before its execution result is recorded.
-func (ht *HookTracker) Add(podNamespace, podName, container, source, hookName string, hookPhase hookPhase) {
+// Currently, the backup hookTracker only tracks a single backup, allowing the name to be set to empty string for simplicity."
```

Comment:
I don't think we should allow using the tracker both for multiple CRs and at the same time one CR.